### ISSUE TITLE
Medusa-V1 Training

### ISSUE THREAD

**Issue by ByteSizedBandit** [aiola-lab/whisper-medusa#24]
Hi,

I’m currently fine-tuning the whisper-medusa-v1 model, aiming for a lossless variant as described in the Medusa paper. Here are the parameters I used for training on Librispeech:

`--gradient-accumulation-steps 2`
`--max-steps 20000`
`--batch-size 8`
`--seed 42`
`--warmup-steps 0`
`--language en`
`--fp16 False`
`--lr 1e-4`
`--optim adafactor`
`--lr-scheduler-type constant`
`--predict-with-generate True`
`--whisper-model-name openai/whisper-large-v2`
`--parts-to-freeze whisper`
`--medusa_num_layers 1`
`--medusa-num-heads 10`
`--medusa-hidden-size 1280`
`--medusa-choices 1,1,1,1,1,1,1,1,1,1,1`
`--medusa-heads-type base_head`
`--medusa-loss-on-original False`
`--medusa-kl-loss False`
`--medusa-kl-weight 0`
`--output-whisper-original False`

Despite following these settings, the trained model does not seem to be lossless. I noticed that your code appears to use the tree decoding scheme by default. Should I somehow disable this to achieve a lossless output for medusa-v1? I’m eager to train a lossless variant as highlighted in the Medusa paper.

Could you please provide guidance on this?

Thank you!

**Comment by YaelSegalFeldman** [comment]
Hi @ByteSizedBandit ,

The completely lossless variant is medusa_block, but there are some mistakes in your running parameters. You should update the following parameters:
--medusa-heads-type=medusa_block
--fp16=true (we use that)
--max-steps=200000 (the number of steps should be set to 200k and not 20k).
Additionally, I recommend using --predict-with-generate=false for improved training speed.
